.github/workflows/label-pr-author.yaml

```diff
@@ -0,0 +1,46 @@
+name: Label PR Author as Internal or External
+
+on:
+  pull_request:
+    types: [opened, reopened, ready_for_review, synchronize, labeled, unlabeled]
+
+permissions:  
+  pull-requests: write
+  issues: write       
+  members: read
+
+jobs:
+  label-author-type:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Auto-label PR author
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const pr = context.payload.pull_request;
+            const labels = pr.labels.map(label => label.name.toLowerCase());
+
+            const hasLabel = labels.includes("maintainer") || labels.includes("contributor");
```

Comment:
Should this use uppercase labels instead?

Reply:
sure, we can do that, would make it one line less I suppose

Reply:
Yes, that'd be better. Thanks. Pushed changes.